### PR TITLE
shanoir-issue#1316 Download dataset naming - special chars on windows

### LINKED ISSUE #1316: On windows, download with special characters (like *) does not work

### PR DESCRIPTION
**Pull request by jcomedouteau**
How to test:
- rename a dataset, examination, subject with special chars (/!?^() etc..) and download one and multiple datasets.
- Check that it works on windows. 
Closes #1316

### CODE REVIEW

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/download/WADODownloaderService.java [hunk 1]
```diff
@@ -97,6 +97,8 @@ public class WADODownloaderService {
 
 	private static final String CONTENT_TYPE = "&contentType";
 	
+	private static final char[] FORBIDDEN_FILENAME_CHARS = {'\\', '/', ':', '*', '?', '\"', '<', '>', '|'};
```
Comment (Nunien) [inline shanoir-ng-datasets/src/main/java/org/shanoir/ng/download/WADODownloaderService.java:100]: Is this used ?

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/controler/DatasetApiController.java [hunk 2]
```diff
@@ -346,9 +346,9 @@ public void downloadDatasetById(
 		if (dataset.getUpdatedMetadata() != null && dataset.getUpdatedMetadata().getComment() != null) {
 			datasetName += "-" + dataset.getUpdatedMetadata().getComment();
 		}
-		if (datasetName.contains(File.separator)) {
-			datasetName = datasetName.replaceAll(File.separator, "_");
-		}
+		// Replace all forbidden characters.
+		datasetName = datasetName.replaceAll("[^a-zA-Z0-9\\.\\-]", "_");
```
Comment (Nunien) [inline shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/controler/DatasetApiController.java:350]: what does it replace ?

**Review by Nunien — CHANGES_REQUESTED** [review]

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/download/WADODownloaderService.java (hunk 1 above)
Comment (jcomedouteau) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/download/WADODownloaderService.java:100]: Not anymore, I'm removing it.

**Comment by jcomedouteau** [comment]
I removed the useless line and added a test to check the code.

**Review by julien-louis — APPROVED** [review]

**Pull request merged**